Tests/LoggerMessageStringTests.swift

```diff
@@ -0,0 +1,466 @@
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 1, "Actual number of writes should be 1")
+    }
+
+    func testThatItLogsAsExpectedWithInfoLogLevel() {
+        // Given
+        let (log, writer) = logger(logLevels: .info)
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 1, "Actual number of writes should be 1")
+    }
+
+    func testThatItLogsAsExpectedWithEventLogLevel() {
+        // Given
+        let (log, writer) = logger(logLevels: .event)
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 1, "Actual number of writes should be 1")
+    }
+
+    func testThatItLogsAsExpectedWithWarnLogLevel() {
+        // Given
+        let (log, writer) = logger(logLevels: .warn)
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 1, "Actual number of writes should be 1")
+    }
+
+    func testThatItLogsAsExpectedWithErrorLogLevel() {
+        // Given
+        let (log, writer) = logger(logLevels: .error)
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 1, "Actual number of writes should be 1")
+    }
+
+    func testThatItLogsAsExpectedWithAllLogLevel() {
+        // Given
+        let (log, writer) = logger(logLevels: .all)
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 5, "Actual number of writes should be 5")
+    }
+
+    func testThatItLogsAsExpectedWithMultipleLogLevels() {
+        // Given
+        let logLevels: LogLevel = [.debug, .event, .error]
+        let (log, writer) = logger(logLevels: logLevels)
+
+        // When
+        log.debug { "" }
+        log.info { "" }
+        log.event { "" }
+        log.warn { "" }
+        log.error { "" }
+
+        // Then
+        XCTAssertEqual(writer.actualNumberOfWrites, 3, "Actual number of writes should be 3")
+    }
+
+    func testThatItCanLogMessageInsideAnotherLogStatement() {
+        // Given
+        let (log, writer) = logger(logLevels: .all)
+
+        // When
+        log.debug { () -> String in
+            log.debug { "" }
+            return ""
+        }
```

Comment:
Pretty sure you have to keep the signature here right? Otherwise the compiler freaks out?